internal/unit/file.go

```diff
@@ -84,7 +84,11 @@ func (u *File) String() string {
 }
 
 // Insert adds name=value to section and returns a newly parsed pointer to File.
+// If name is the empty string this is a noop.
 func (u *File) Insert(section, name string, value ...string) *File {
+	if name == "" {
```

Comment:
Please, add a comment above stating that this is empty only when the option is not supported by the current systemd and no alternative is available.